Tests/Source/RangeTester.cpp

```diff
@@ -3,24 +3,18 @@
 #include <catch.hpp>
 #include "Rover/Constant.hpp"
 #include "Rover/Generator.hpp"
+#include "Rover/Noncopyable.hpp"
 #include "Rover/Range.hpp"
+
 using namespace Rover;
 
 namespace {
 
```

Comment:
Remove blank line on 12.

Reply:
Is it a special case for unnamed namespaces? We tend to put a blank line after namespace Rover { for example. What about namespace Details?

Reply:
The rule is that a blank line preceeds all documentation, ie.

```
namespace Foo {
  <blank line>
  /** some documentation. */
  class xxx {};
```

That blank line is there because of the documentation, which must always be preceded by a blank line no matter what. If there's no documentation then the blank line is removed.

I know this is a nuisance and should be better documented in the style guide...

Reply:
Okay. If we look at Range.hpp though, Range constructors are documented while generate function is not. Should the blank line in between them be removed then? Or should any documented function/object/type/whatever be isolated by blank lines on both sides?